software/MainboardCommunication.py

```diff
@@ -0,0 +1,17 @@
+import struct
```

Comment:
Avoid creating task specific files. Instead structure the code as it would work in the full robot code. For mainboard communication you can create a class that deals with opening the serial port and sending commands to mainboard. Then use mainboard communication class in main robot class or file.